### PR TITLE
Fix search driver hooks

### PR DESCRIPTION
**Pull request by joemcelroy**
To fix the typescript issues raised in #687.

### CODE REVIEW

**Comment by elasticdocs** [comment]
:rocket: Built _search-ui_ successfully!

* [View deploy preview](https://search-ui-5zx06l8fz-elastic-dev.vercel.app)

* Built with commit ba9b07d8a5dbba889003f2abf594b4d6b65a2b32

> Issues? Visit #next-docs in Slack

File: packages/search-ui/src/Events.ts [hunk 1]
```diff
@@ -43,25 +50,25 @@ function wireUpEventHandler(
...
   public autocomplete: (
-    query: { searchTerm: string },
-    queryConfig: AutocompleteQuery
-  ) => Promise<any>;
-  public resultClick: () => void;
-  public autocompleteResultClick: () => void;
+    query: AutocompleteSearchQuery,
+    queryConfig: AutocompleteQueryConfig
+  ) => Promise<AutocompleteResponseState>;
+  public resultClick: (resultParams: any) => void;
```
Comment (joemcelroy) [inline packages/search-ui/src/Events.ts:70]: types need addressing here but decided to ignore this for now until this PR has been merged.

File: packages/search-ui-workplace-search-connector/src/__tests__/WorkplaceSearchAPIConnector.test.ts [hunk 2]
```diff
@@ -98,7 +99,7 @@ describe("WorkplaceSearchAPIConnector", () => {
         beforeAutocompleteSuggestionsCall?: SuggestionsQueryHook;
       } = {}
     ) {
-      if (!state.searchTerm) state.searchTerm = "searchTerm";
+      if (!state || !!state.searchTerm) state.searchTerm = "searchTerm";
```
Comment (yakhinvadim) [inline packages/search-ui-workplace-search-connector/src/__tests__/WorkplaceSearchAPIConnector.test.ts:102]: Should it be `!state.searchTerm`? (one `!`)

**Review by yakhinvadim — APPROVED** [review]
Nice improvements Joe! Just 1 question:

File: packages/search-ui-workplace-search-connector/src/__tests__/WorkplaceSearchAPIConnector.test.ts (hunk 2 above)
Comment (joemcelroy) [reply packages/search-ui-workplace-search-connector/src/__tests__/WorkplaceSearchAPIConnector.test.ts]: thanks for spot! Should in fact be removed as got a default object for state in the arguments and this line was never used.

**Review by yakhinvadim — APPROVED** [review]
🎉

**Pull request merged**